tests/Aspire.Dashboard.Tests/Model/SpanWaterfallViewModelTests.cs

```diff
@@ -40,4 +41,82 @@ public void Create_HasChildren_ChildrenPopulated()
                 Assert.Empty(e.Children);
             });
     }
+
+    [Theory]
+    [InlineData("1234", true)]  // Matches span ID
+    [InlineData("app1", true)]  // Matches application name
+    [InlineData("Test", true)]  // Matches display summary
+    [InlineData("peer-service", true)]  // Matches uninstrumented peer
+    [InlineData("nonexistent", false)]  // Doesn't match anything
+    public void MatchesFilter_VariousCases_ReturnsExpected(string filter, bool expected)
```

Comment:
Could you add a couple more tests:

* Parent spans are included when a child is matched
* Child spans are included when a parent is matched